packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java

```diff
@@ -284,10 +279,9 @@ ResponseEntity<Dataset> updateDataset(
 			@ApiResponse(responseCode = "500", description = "There was an issue retrieving the dataset from the data store", content = @Content)
 	})
 	public ResponseEntity<CsvAsset> getCsv(
-			@PathVariable("id") final UUID datasetId,
-			@RequestParam("filename") final String filename,
-			@RequestParam(name = "limit", defaultValue = "-1", required = false) final Integer limit // -1 means no
-																										// limit
+		@PathVariable("id") final UUID datasetId,
+		@RequestParam("filename") final String filename,
+		@RequestParam(name = "limit", defaultValue = "" + DEFAULT_CSV_LIMIT, required = false) final Integer limit
```

Comment:
We should probably also stop the CSV file from being loaded into RAM in its entirety - this will stop the client from getting all